plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/TrinoS3FileSystem.java

```diff
@@ -605,16 +612,32 @@ public boolean isFilesOnly()
         }
     }
 
-    private Iterator<LocatedFileStatus> listPrefix(Path path, OptionalInt initialMaxKeys, ListingMode mode)
+    /**
+     * List all objects rooted at the provided path.
+     */
+    private Iterator<LocatedFileStatus> listPath(Path path, OptionalInt initialMaxKeys, ListingMode mode)
     {
         String key = keyFromPath(path);
         if (!key.isEmpty()) {
             key += PATH_SEPARATOR;
         }
 
+        return listPrefix(key, initialMaxKeys, mode);
+    }
+
+    /**
+     * List all objects whose absolute path matches the provided prefix.
+     */
+    private Iterator<LocatedFileStatus> listPrefix(Path prefix, OptionalInt initialMaxKeys, ListingMode mode)
+    {
+        return listPrefix(keyFromPath(prefix), initialMaxKeys, mode);
+    }
+
+    private Iterator<LocatedFileStatus> listPrefix(String prefix, OptionalInt initialMaxKeys, ListingMode mode)
```

Comment:
It looks like this is only used by the above method, so we should just inline it.

Reply:
This one is used in two places, but the above one (`listPrefix(Path prefix, ..`) is a good candidate for inline and it seems this is what @hashhar did inline